hyperjumptech/monika
# Make timeout explicit for all probe and http requests

# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Closing #907 

## How did you implement / how did you fix it  
1.  Made all requests have explicits timeout
2. Route all axios request through a utils sendHttpRequest()
3. Add timeout as default on axios.config

reference: https://medium.com/@masnun/handling-timeout-in-axios-479269d83c68

## How to test
1. run `npm run test` should all pass
2. All existing probe types should work
3. Add various notifications that pushes requests to a url, notifications should all work.

## Screenshot

ping test works:
![image](https://user-images.githubusercontent.com/964106/195865814-76fcd6e0-c18f-4c0b-b577-19b51ce51de0.png)

url get requests works:
![image](https://user-images.githubusercontent.com/964106/195866226-64fa553e-cb13-41a2-a7b6-c75d06a29a2d.png)

notifications to ms-teams successful:
![image](https://user-images.githubusercontent.com/964106/195868340-6d6b6cfa-a245-47b4-a282-bfa713015a2d.png)
